### ISSUE TITLE
User creation not working

### ISSUE THREAD

**Issue by Galexrt** [Galexrt/docker-zulip#16]
> But I meet another problem.
> The 'forget password' email can't be sent.
> I use the mailgun to send email, and it shows that zulip does not calling mailgun to send email.
> 
> Also, I can't login zulip by the ZULIP_USER_EMAIL and ZULIP_USER_PASSWORD, do I miss something?
> - @njuaplusplus

This has to do with this script, that get's executed once "during" the initiation [zulip-puppet/files/setupZulipUser.sh](zulip-puppet/files/setupZulipUser.sh)

**Comment by njuaplusplus** [comment]
@Galexrt 
How do you login zulip?
By which user and password?

**Comment by Galexrt** [comment]
@njuaplusplus If you haven't changed the env vars `ZULIP_USER_EMAIL` and `ZULIP_USER_PASS`.
The login would be, username/email: `example@example.com`, with password `zulip`.

These two env vars control the first user `ZULIP_USER_EMAIL` and `ZULIP_USER_PASS`.
**They will only one time work!**

**Comment by Galexrt** [comment]
**In the v1.3.7 version it is currently not working!**
Please try with v1.3.7 and using the registration page to create users again. Thanks!

I try to re-add this feature. I close this ticket for now and create a new ticket for re-adding/fixing this feature.

**Comment by njuaplusplus** [comment]
@Galexrt 
I will try it.

As you say 'They will only one time work!'
Every time I test your image, I first remove the whole zulip directory, including other volumns,
and run from the scratch.
So it may not be the creation of user for the second time.

**Comment by njuaplusplus** [comment]
Failed to login......

I think maybe that the user is not created, that's why the 'forget password' email is not sent...

But also I failed to register...

> Your e-mail does not match any existing open organization. Use a different e-mail address, or contact aplusplus@qq.com with questions.

And If I choose to sign up with google account, it says 500 error.

**Comment by Galexrt** [comment]
@njuaplusplus The behaviour changed in zulip version 1.3.7.
The user creation has to be done using the zulip registration page. My script that did it, isn't working since the Zulip version 1.3.7, but I'll try my best to fix the script soon.

I haven't tested google login yet, but I'll look into it when I have fixed most of the bugs.

In version 1.3.6 `ZULIP_USER_EMAIL`, `ZULIP_USER_DOMAIN` and `ZULIP_SETTINGS_ADMIN_DOMAIN` had to be from the same domain. (As I mentioned the user creation per my script, isn't working currently)

**Comment by Galexrt** [comment]
@njuaplusplus The registration error you describe means, that for example my Zulip Admin domain is set to "example.com" so you can only create accounts with email addresses from "example.com".

**Google Login**: Have you set the needed Zulip Google Login vars?
See https://github.com/zulip/zulip/blob/master/zproject/local_settings_template.py and https://github.com/zulip/zulip/blob/master/zproject/settings.py

**Comment by njuaplusplus** [comment]
@Galexrt 
It's my fault that I have not set the google vars.

Maybe it's better to add this restriction in the readme.md:
> ZULIP_USER_EMAIL, ZULIP_USER_DOMAIN and ZULIP_SETTINGS_ADMIN_DOMAIN had to be from the same domain

**Comment by Galexrt** [comment]
@njuaplusplus I added a wiki page for it ;)
Here https://github.com/Galexrt/docker-zulip/wiki/Zulip-GoogleApps-Authentication

**Comment by njuaplusplus** [comment]
@Galexrt 

Thank you so much for your work.

But I think that there's some misleading at the
[configuration wiki page](https://github.com/Galexrt/docker-zulip/wiki/Configuration):

> ZULIP_USER_DOMAIN: User domain, should be same as used in ZULIP_SETTINGS_ADMIN_DOMAIN or ZULIP_SETTINGS_EXTERNAL_HOST

The `ZULIP_SETTINGS_EXTERNAL_HOST` can be different.

As in my vps, I set the `ZULIP_USER_DOMAIN="qq.com"`,
but the `ZULIP_SETTINGS_EXTERNAL_HOST="zulip.answ.me"`.
And it works.

For anyone want to try zulip, feel free to browser the site.

By the way, have you deployed zulip on your vps for external access?

**Comment by Galexrt** [comment]
@njuaplusplus Yes you are right, I try to make this more clear in the wiki.
Yes, I have deployed docker-zulip to one of my machines, with external access.

If you have any other questions feel free to ask them on the Freenode IRC #docker-zulip (See https://github.com/Galexrt/docker-zulip/blob/master/README.md#community for more details) or here too.
Closing as the issue is going to be resolved in #25.
